core/src/main/java/org/kiwix/kiwixmobile/core/compat/ResolveInfoCompat.kt

```diff
@@ -0,0 +1,164 @@
+ * [android.os.StrictMode.VmPolicy.Builder.detectImplicitDirectBoot]
+ * to detect when a caller is relying on implicit automatic matching,
+ * instead of confirming the explicit behavior they want, using a
+ * combination of these flags:
+ *
+ *  * [PackageManager.MATCH_DIRECT_BOOT_AWARE]
+ *  * [PackageManager.MATCH_DIRECT_BOOT_UNAWARE]
+ *  * [PackageManager.MATCH_DIRECT_BOOT_AUTO]
+ *
+ */
+const val MATCH_DIRECT_BOOT_AUTO = 0x10000000
+
+/**
+ * Querying flag: match components which are direct boot *aware* in
+ * the returned info, regardless of the current user state.
+ *
+ *
+ * When neither [.MATCH_DIRECT_BOOT_AWARE] nor
+ * [.MATCH_DIRECT_BOOT_UNAWARE] are specified, the default behavior is
+ * to match only runnable components based on the user state. For example,
+ * when a user is started but credentials have not been presented yet, the
+ * user is running "locked" and only [.MATCH_DIRECT_BOOT_AWARE]
+ * components are returned. Once the user credentials have been presented,
+ * the user is running "unlocked" and both [.MATCH_DIRECT_BOOT_AWARE]
+ * and [.MATCH_DIRECT_BOOT_UNAWARE] components are returned.
+ *
+ * @see UserManager.isUserUnlocked
+ */
+const val MATCH_DIRECT_BOOT_AWARE = 0x00080000
+
+/**
+ * Querying flag: match components which are direct boot *unaware* in
+ * the returned info, regardless of the current user state.
+ *
+ *
+ * When neither [.MATCH_DIRECT_BOOT_AWARE] nor
+ * [.MATCH_DIRECT_BOOT_UNAWARE] are specified, the default behavior is
+ * to match only runnable components based on the user state. For example,
+ * when a user is started but credentials have not been presented yet, the
+ * user is running "locked" and only [.MATCH_DIRECT_BOOT_AWARE]
+ * components are returned. Once the user credentials have been presented,
+ * the user is running "unlocked" and both [.MATCH_DIRECT_BOOT_AWARE]
+ * and [.MATCH_DIRECT_BOOT_UNAWARE] components are returned.
+ *
+ * @see UserManager.isUserUnlocked
+ */
+const val MATCH_DIRECT_BOOT_UNAWARE = 0x00040000
+
+@LongDef(
+  flag = true,
+  // prefix = ["GET_", "MATCH_"],
+  value = [
+    GET_META_DATA.toLong(),
+    GET_RESOLVED_FILTER.toLong(),
+    GET_SHARED_LIBRARY_FILES.toLong(),
+    MATCH_ALL.toLong(),
+    MATCH_DISABLED_COMPONENTS.toLong(),
+    MATCH_DISABLED_UNTIL_USED_COMPONENTS.toLong(),
+    MATCH_DEFAULT_ONLY.toLong(),
+    MATCH_DIRECT_BOOT_AUTO.toLong(),
+    MATCH_DIRECT_BOOT_AWARE.toLong(),
+    MATCH_DIRECT_BOOT_UNAWARE.toLong(),
+    MATCH_SYSTEM_ONLY.toLong(),
+    MATCH_UNINSTALLED_PACKAGES.toLong()
+    // PackageManager.MATCH_INSTANT, // @SystemApi
+    // PackageManager.MATCH_DEBUG_TRIAGED_MISSING, // deprecated
+    // PackageManager.GET_DISABLED_COMPONENTS, // deprecated
+    // PackageManager.GET_DISABLED_UNTIL_USED_COMPONENTS, // deprecated
+    // PackageManager.GET_UNINSTALLED_PACKAGES.toLong() // deprecated
+  ]
+)
+@Retention(AnnotationRetention.SOURCE)
+annotation class ResolveInfoFlagsBitsCompat
```

Comment:
this class too.

Reply:
@criticalAY , you forgot to make this class in a separate file.

Reply:
there is an error if we don't dont declare them in same file

Reply:
we should fix this.

Reply:
> i am facing a top level declaration error while committing the code it's not shown in Android studio though

I see now that the class name is ```ResolveInfoFlagsCompat``` and the file name is ```ResolveInfoCompat```. This is why the error is occurring. To resolve it, you can either rename the file or rename the class. Either way, this should fix the issue.